parquet-variant-compute/src/variant_parser.rs

```diff
@@ -0,0 +1,669 @@
+        let is_large = (value_header & 0x10) != 0;
+
+        let num_elements_size = if is_large { 4 } else { 1 };
+        let element_offset_size = (element_offset_size_minus_one + 1) as usize;
+
+        Ok(ArrayHeader {
+            num_elements_size,
+            element_offset_size,
+            is_large,
+        })
+    }
+
+    /// Unpack integer from bytes
+    pub fn unpack_int(bytes: &[u8], size: usize) -> Result<usize, ArrowError> {
+        if bytes.len() < size {
+            return Err(ArrowError::InvalidArgumentError(
+                "Not enough bytes to unpack integer".to_string(),
+            ));
+        }
+
+        match size {
+            1 => Ok(bytes[0] as usize),
+            2 => Ok(u16::from_le_bytes([bytes[0], bytes[1]]) as usize),
+            3 => Ok(u32::from_le_bytes([bytes[0], bytes[1], bytes[2], 0]) as usize),
+            4 => Ok(u32::from_le_bytes([bytes[0], bytes[1], bytes[2], bytes[3]]) as usize),
+            _ => Err(ArrowError::InvalidArgumentError(format!(
+                "Invalid integer size: {}",
+                size
+            ))),
+        }
+    }
+
+    /// Calculate the size needed to store an integer
+    pub fn calculate_int_size(value: usize) -> usize {
+        if value <= u8::MAX as usize {
+            1
+        } else if value <= u16::MAX as usize {
+            2
+        } else if value <= 0xFFFFFF {
+            3
+        } else {
+            4
+        }
+    }
+
+    /// Build object header byte
+    pub fn build_object_header(
+        is_large: bool,
+        field_id_size: usize,
+        field_offset_size: usize,
+    ) -> u8 {
+        let large_bit = if is_large { 1 } else { 0 };
+        (large_bit << 6)
+            | (((field_id_size - 1) as u8) << 4)
+            | (((field_offset_size - 1) as u8) << 2)
+            | 2
+    }
+
+    /// Write integer bytes to buffer
+    pub fn write_int_bytes(buffer: &mut Vec<u8>, value: usize, size: usize) {
+        match size {
+            1 => buffer.push(value as u8),
+            2 => buffer.extend_from_slice(&(value as u16).to_le_bytes()),
+            3 => {
+                let bytes = (value as u32).to_le_bytes();
+                buffer.extend_from_slice(&bytes[..3]);
+            }
+            4 => buffer.extend_from_slice(&(value as u32).to_le_bytes()),
+            _ => panic!("Invalid size: {}", size),
+        }
+    }
+
+
+
+    /// Check if value bytes represent a primitive
+    pub fn is_primitive(value_bytes: &[u8]) -> bool {
+        if value_bytes.is_empty() {
+            return false;
+        }
+        Self::get_basic_type(value_bytes[0]) == VariantBasicType::Primitive
+    }
+
+    /// Check if value bytes represent a short string
+    pub fn is_short_string(value_bytes: &[u8]) -> bool {
+        if value_bytes.is_empty() {
+            return false;
+        }
+        Self::get_basic_type(value_bytes[0]) == VariantBasicType::ShortString
```

Comment:
nit:
```suggestion
        value_bytes.get(0).is_some_and(|header| {
            Self::get_basic_type(header) == VariantBasicType::ShortString
        })
```
(more below, if you like this idea)